src/app/controllers/__tests__/frontend.server.controller.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { StatusCodes } from 'http-status-codes'
+
+import expressHandler from 'tests/unit/backend/helpers/jest-express'
+
+import frontendServerController from '../frontend.server.controller'
+
+describe('frontend.server.controller', () => {
+  afterEach(() => jest.clearAllMocks())
+  const mockRes = expressHandler.mockResponse()
+
+  it('should return the correct response when the request is valid', () => {
+    const mockReq = {
+      app: {
+        locals: {
+          GATrackingID: 'abc',
+          appName: 'xyz',
+          environment: 'efg',
+        },
+      },
+      query: {
+        redirectPath: 'formId?fieldId1=abc&fieldId2=xyz',
+      },
+    }
+    // datalayer
+    frontendServerController.datalayer(mockReq, mockRes)
+    expect(mockRes.send).toHaveBeenCalledWith(
+      expect.stringContaining("'app_name': 'xyz'"),
+    )
+    expect(mockRes.send).toHaveBeenCalledWith(
+      expect.stringContaining("'config', 'abc'"),
+    )
+    expect(mockRes.type).toHaveBeenCalledWith('text/javascript')
+    expect(mockRes.status).toHaveBeenCalledWith(StatusCodes.OK)
+
+    // environment
+    frontendServerController.environment(mockReq, mockRes)
```

Comment:
wait I think we've gone to the other extreme here haha - each`describe` should only test a single function, so you should have nested `describe` blocks for `datalayer`, `environment` and `redirectLayer`

Reply:
got it, see if edits are ok